src/foam/mlang/mlang.js

```diff
@@ -3480,6 +3480,43 @@ return true;
   ]
 });
 
+foam.CLASS({
+  package: 'foam.mlang.predicate',
+  name: 'HasPermission',
+  extends: 'foam.mlang.predicate.Binary',
```

Comment:
Should this extend Binary? It isn't a normal Binary mlang, in that each argument isn't a Function, it just has two arguments. Then the args could be properly typed.

Reply:
That was something I wasn't sure about. Should it just extend `AbstractPredicate` then?

Reply:
Resolved by 65ef3a0.